From 6a0f9d144c71d3450094faf031f604ce3c1a12a6 Mon Sep 17 00:00:00 2001
From: Adam Borowski <kilobyte@angband.pl>
Date: Thu, 5 Apr 2018 19:45:11 +0200
Subject: [PATCH 1/3] Check if in-target /bin/true works before more complex
 stuff.

The first command of second stage used to be mount /proc, which has plenty
of other reasons to fail, and people tend to try those.  Instead, check
first if in-target binaries are executable at all.  They can fail because
of arch unsupported by the machine (and no qemu), missing or borked libc,
borked ld-linux -- so let's separate those from mount fails.
---
 scripts/debian-common | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/debian-common b/scripts/debian-common
index 36989a2..4ab1fe8 100644
--- a/scripts/debian-common
+++ b/scripts/debian-common
@@ -59,6 +59,8 @@ first_stage_install () {
 }
 
 second_stage_install () {
+	in_target /bin/true
+
 	setup_dynamic_devices
 
 	x_feign_install () {
-- 
2.17.0

